pallets/gear/src/lib.rs

```diff
@@ -1665,7 +1682,8 @@ pub mod pallet {
             <T as Config>::Currency::reserve(&who, gas_limit_reserve)
                 .map_err(|_| Error::<T>::NotEnoughBalanceForReserve)?;
 
-            let _ = GasHandlerOf::<T>::create(origin.clone(), message_id, gas_limit);
+            GasHandlerOf::<T>::create(origin.clone(), message_id, gas_limit)
+                .map_err(|_| Error::<T>::MessageAlreadyReplied)?;
```

Comment:
this doesn't make sense, due to replied message was already remove from mailbox, so we will fail earlier

Reply:
what about

```rust
  // # Safety                                                                                                                                                                                                                                                         
  //                                                                                                                                                                                                                                                                  
  // This is unreachable since the `message_id` is new generated                                                                                                                                                                                                      
  // from `original_message` by our system, and `original_message`
  // has been just removed.
```

Reply:
Original message wasn't used to generate new message id, because otherwise there should be no message in mailbox and the extrinsic fails earlier

Reply:
But we fail here if message was already replied or removed from mailbox

https://github.com/gear-tech/gear/blob/b4f017fdaa20b5e7e4074a9848dbe37145bc30c8/pallets/gear/src/lib.rs#L1632